File: interpreter/typescript/accounting-expression.ts
import {IJobExpression} from "./job-expression.interface";
import {Context} from "./context";

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
export class AccountingExpression implements IJobExpression {
  public interpret(context: Context): void {
    if(context.formula.includes('M')) {
      context.totalPoint += 3000;
    }
  }
}

File: interpreter/go/accounting-expression.go
package main

import "strings"

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
type accountingExpression struct{}

func NewAccountingExpression() *accountingExpression {
	return &accountingExpression{}
}

func (e accountingExpression) Interpret(ctx *Context) {
	if strings.Contains(ctx.Formula, "M") {
		ctx.TotalPoint += 3000
	}
}
